### PR TITLE
fix for new 'time_iso' variable in GFS history files

### PR DESCRIPTION
**Pull request by jswhit**
https://github.com/ufs-community/ufs-weather-model/issues/1052 added a new 'time_iso' character array variable in the GFS history files.  This PR fixes NCEPLIBS-ncio so it correctly handles this variable.   Without this fix, the netcdf library will exit with 'NetCDF start count exceeds dimension bound' error code.  The fix involves always passing start and count arrays to nf90_put_var and nf90_get_var since the failure only occurs when these optional arguments are not given and the fortran interface infers them from the variable dimensions.  The tests have also been updated.

@CoryMartin-NOAA

### CODE REVIEW

File: tests/tst_ncio.F90 [hunk 1]
```diff
@@ -24,11 +26,11 @@ program tst_ncio
   print *,'*** Test creation of new dataset from template...'
   dset = create_dataset('dynf000.nc',dsetin)
   print *,'*** Test that number of variables,dimensions,attributes is read...'
-  if (dsetin%nvars .ne. 24) then
+  if (dsetin%nvars .ne. 25) then
```
Comment (kgerheiser) [inline tests/tst_ncio.F90]: Could you make these constants? As bare number its hard to tell what they mean.

File: tests/tst_ncio.F90 [hunk 2]
```diff
@@ -24,11 +26,11 @@ program tst_ncio
...
      print *,'***number of variables not correct...'
      stop 99
   endif
-  if (dsetin%ndims .ne. 6) then
```
Comment (kgerheiser) [inline tests/tst_ncio.F90:31]: Same here

Comment (jswhit) [reply tests/tst_ncio.F90:31]: Done

File: tests/tst_ncio.F90 (hunk 1 above)
Comment (jswhit) [reply tests/tst_ncio.F90]: Done

Comment (kgerheiser) [reply tests/tst_ncio.F90]: Thank you

File: tests/tst_ncio_mpi.F90 [hunk 3]
```diff
@@ -45,8 +45,8 @@ end subroutine check
 
   if (my_rank .eq. 0) print *,'*** Test that number of variables,dimensions,attributes is read...'
 
-  if (dsetin%nvars .ne. 24) stop 5
-  if (dsetin%ndims .ne. 6) stop 6
+  if (dsetin%nvars .ne. 25) stop 5
+  if (dsetin%ndims .ne. 7) stop 6
```
Comment (kgerheiser) [inline tests/tst_ncio_mpi.F90]: Could you apply those constants to this file too? Might require separating it out into its own file.

Comment (jswhit) [reply tests/tst_ncio_mpi.F90]: Done

**Comment by jswhit** [comment]
the macosx CI test fails when building netcdf-fortran, because the library "-lzip" is not found.  Don't know how to fix this.

**Comment by kgerheiser** [comment]
I'll take a look

**Comment by aerorahul** [comment]
Can we get a move on this PR?  The fix in this PR is required in several DA related utilities required for cycling with the latest UFS-weather-model.

**Comment by kgerheiser** [comment]
#55 should fix the automated build, and then if you could update your PR with those changes this can be merged.

**Comment by kgerheiser** [comment]
@jswhit can you remove the changes to the workflow action?

**Comment by jswhit** [comment]
> @jswhit can you remove the changes to the workflow action?

Done

**Review by kgerheiser — APPROVED** [review]

**Pull request merged**